src/components/partnership/dto/partnership.dto.ts

```diff
@@ -0,0 +1,39 @@
+import { GraphQLString } from 'graphql';
+import { Field, ObjectType } from 'type-graphql';
+import {
+  DateTimeField,
+  Resource,
+  SecuredDateTime,
+  SecuredProperty,
+} from '../../../common';
+import { Organization } from '../../organization';
+import { PartnershipAgreementStatus } from './partnership-agreement-status.enum';
+import { PartnershipType } from './partnership-type.enum';
+
+@ObjectType({
+  description: SecuredProperty.descriptionFor('a partnership agreement status'),
+})
+export abstract class SecuredPartnershipAgreementStatus extends SecuredProperty<
+  PartnershipAgreementStatus
+>(GraphQLString) {}
+
+@ObjectType()
+export class Partnership extends Resource {
+  @Field()
+  readonly agreementStatus: SecuredPartnershipAgreementStatus;
+
+  @Field()
+  readonly mouStatus: SecuredPartnershipAgreementStatus;
+
+  @DateTimeField()
+  readonly mouStart: SecuredDateTime;
+
+  @DateTimeField()
+  readonly mouEnd: SecuredDateTime;
+
+  @Field(() => Organization, { nullable: true })
+  readonly organization: Organization | null;
+
+  @Field(() => [PartnershipType])
+  readonly types: PartnershipType[];
```

Comment:
I think this needs to be secured as well, so that the frontend can show it's readonly. This is a list but it can be treated as a "scalar/single" value for this case.

Reply:
Got it! I'm running into a similar issue here as I am with `SecuredPartnershipAgreementStatus` - can't seem to easily create a `SecuredProperty` that returns a secured `value` of `PartnershipType[]` - is that the right approach?

Reply:
Bleh. Yeah that is the right output. Sounds like for this case we should just forgo the `SecuredProperty` abstraction. See `SecuredDate` for implementation without using abstraction.

Reply:
Okay, I took your advice but went a slightly different direction and created a new abstract class: `SecuredPropertyList`. It takes the same `T` types as `SecuredProperty`, only they're all arrays. Check it out in `secured-property.ts`: 

```typescript

export function SecuredPropertyList<T>(
  ValueClassList:
    | Array<ClassType<T>>
    | Array<AbstractClassType<T>>
    | GraphQLScalarType[]
    | object[]
) {
  @ObjectType({ isAbstract: true, implements: [Readable, Editable] })
  abstract class SecuredPropertyClass
    implements Readable, Editable, Secured<T> {
    @Field(() => ValueClassList, { nullable: true })
    readonly value?: T;
    @Field()
    readonly canRead: boolean;
    @Field()
    readonly canEdit: boolean;
  }

  return SecuredPropertyClass;
}
```

Seems to get us what we need and allows us to have more secured lists of different types in the future. What do you think of this approach? 

If this is too much abstraction for something that you believe will only ever happen once or twice, I could implement it locally in the partnership.dto instead following the `SecuredDate` pattern - lemme know!

Reply:
Now that I'm looking at this more, I'm thinking I could easily just add array types to SecuredProperty as well. Lemme know your thoughts on either approach!

Reply:
I think this is a good direction. Realizing now we'll have more of these properties so this will be useful.

One tweak though, still pass in a single value/type and map it to a list in this abstraction.
```typescript
  abstract class SecuredPropertyClass
    implements Readable, Editable, Secured<T[]> {
    @Field(() => [ValueClass], { nullable: true })
    readonly value?: T[];
  }
```

I also wonder if the hidden value should be null/undefined or an empty list? I could see both ways making sense...

Reply:
Perfect, just made those changes!

Generally, I'm a fan of empty non-nullable lists in GQL. But given this is access-related, I can see a distinction between null/undefined vs an empty list. Empty list = there is no data. null/undefined = you don't have access. What do you think?

Reply:
Changes look good!

> Generally, I'm a fan of empty non-nullable lists in GQL. But given this is access-related, I can see a distinction between null/undefined vs an empty list. Empty list = there is no data. null/undefined = you don't have access. What do you think?

Well no because `canRead` indicates whether it's empty with access or not accessible.
I think we should probably be consistent anyways and we do the same thing for `SecuredList`. So let's make it non-nullable.

Reply:
Got it! Makes sense, just pushed up the update making it non-nullable

Reply:
So doing means the `SecuredProperty.descriptionFor` is incorrect for this case.
Can you create a `SecuredPropertyList.descriptionFor` to use instead?
Sorry I know this is a lot of back and forth ☹️

Reply:
Haha no worries at all man. Just pushed up a `SecuredPropertyList.descriptionFor` - lemme know how that looks!

Reply:
Looks great!